src/primitives/line/mod.rs

```diff
@@ -202,19 +191,59 @@ impl Line {
         self.end - self.start
     }
 
-    /// Get which side of a line a point lies on.
+    /// Checks if the point lies on the given side of the line.
     ///
-    /// If the result is zero, the point lies on the line. Otherwise, values greater than zero
-    /// denote that the point is on the left side of the line. Values less than zero denote the
-    /// point lies to the right.
-    pub(in crate::primitives) fn side(&self, point: Point) -> i32 {
+    /// Returns `true` if the point lies on the correct side or on the line.
+    pub(in crate::primitives) fn check_side(&self, point: Point, side: LineSide) -> bool {
```

Comment:
I think the entire body of this fn can be replaced with:

```rust
LinearEquation::from_line(&self).check_side(point, side)
```

Performance for the `thick polyline` bench went up to ~+12% compared to master with this change. `filled triangle` is +~3%.

Reply:
> I think the entire body of this fn can be replaced with:

It can't be replaced at the moment, because `from_line` is only implemented for `LinearEquation<i32>` and `check_side` is only for `LinearEquation<Real>`. But this is something I want to merge in the future.

Reply:
Ah, sorry in my local testing I also changed `LinearEquation<i32>` to:

```rust
impl LinearEquation<i32> {
    pub const fn from_line(line: &Line) -> Self {
        Self {
            a: line.end.y - line.start.y,
            b: line.start.x - line.end.x,
            c: line.end.x * line.start.y - line.start.x * line.end.y,
        }
    }

    pub fn check_side(&self, point: Point, side: LineSide) -> bool {
        let t = self.a * point.x + self.b * point.y + self.c;

        match side {
            LineSide::Left => t >= 0,
            LineSide::Right => t <= 0,
        }
    }
}
```

which IIRC is the only change I needed to make. Wouldn't it make sense to change that in this PR?

Reply:
I could add that change, but I would like to get rid of `LinearEquation<Real>` entirely, which is outside the scope of this PR. So I would prefer to do this in a future PR.

Reply:
Sure thing. Thanks for entertaining my bikeshedding :)